packages/next-app/src/app/pool/[poolId]/(components)/MetadataAttributesTable.tsx

```diff
@@ -111,14 +117,63 @@ function Row({ data }: { data: PoolMetadataAttribute }) {
   );
 }
 
-export function MetadataAttributesTable({ poolId }: { poolId: string }) {
-  const { metadata } = useContext(PoolMetadataContext);
+export function MetadataAttributesTable({ poolId }: { poolId: `0x${string}` }) {
+  const { metadata, handleSetMetadata, handleSetPool } =
+    useContext(PoolMetadataContext);
+
+  handleSetPool(poolId);
+
+  useEffect(() => handleSetMetadata([]), []);
+
+  const { data: pools } = metadataGql.useMetadataPool({
+    poolId,
+  });
+
+  const metadataPool = pools?.pools[0];
+
+  const [poolMetadataCID, setMetadataCID] = useState("");
```

Comment:
any reason to have this value in a state variable given it's already coming from the GraphQL endpoint?